src/meson.build

```diff
@@ -0,0 +1,428 @@
+conf_inc = include_directories('include')
+
+cdata = configuration_data()
+
+cdata.set_quoted('PACKAGE', meson.project_name())
+cdata.set_quoted('VERSION', meson.project_version())
+cdata.set_quoted('CONFDIR', sysconfdir)
+cdata.set_quoted('RUNSTATEDIR', runstatedir)
+cdata.set_quoted('LOCALSTATEDIR', localstatedir)
+cdata.set_quoted('SBINDIR', sbindir)
+
+cdata.set('HAVE_' + os_name.to_upper(), 1)
+
+foreach attr : ['malloc', 'nonnull', 'warn_unused_result']
+  cc.has_function_attribute(attr)
+endforeach
+
+check_headers = [
+  'arpa/inet.h',
+  'arpa/nameser.h',
+  'dirent.h',
+  'getopt.h',
+  'inttypes.h',
+  'net/ethernet.h',
+  'net/if.h',
+  'net/if_arp.h',
+  'net/if_types.h',
+  'netdb.h',
+  'netinet/icmp6.h',
+  'netinet/if_ether.h',
+  'netinet/in.h',
+  'netinet/in6.h',
+  'netinet/in_systm.h',
+  'netinet/ip.h',
+  'netinet/ip6.h',
+  'netinet/ip_icmp.h',
+  'netinet/tcp.h',
+  'netpacket/packet.h',
+  'resolv.h',
+  'stddef.h',
+  'sys/file.h',
+  'sys/ioctl.h',
+  'sys/mman.h',
+  'sys/param.h',
+  'sys/resource.h',
+  'sys/socket.h',
+  'sys/stat.h',
+  'sys/time.h',
+  'sys/types.h',
+  'sys/un.h',
+  'sys/wait.h',
+  'syslog.h',
+  'termios.h',
+]
+
+check_functions = [
+  'asprintf',
+  'daemon',
+  'devname',
+  'fchmod',
+  'fdevname',
+  'fork',
+  'gettimeofday',
+  'mlockall',
```

Comment:
The check for `mlockall()` is failing because `<sys/mman.h>` is not `#include`d while running the test. It has to be added to `src/have.h`. Alternatively, according to the manpage it seems that:

> On POSIX systems on which `mlockall()` and `munlockall()` are available, `_POSIX_MEMLOCK` is defined in `<unistd.h>` to a value greater than 0.  (See also sysconf(3).)

So we could use that instead of checking the function here.